assets/src/blocks/Timeline/useStyleSheet.js

```diff
@@ -0,0 +1,69 @@
+// useScript implementation from: https://usehooks.com/useScript/
+import { useEffect, useState } from 'react';
+
+// Hook
+let cachedStyleSheets = [];
+export const useStyleSheet = (href, media = 'all') => {
+
+  // Keeping track of script loaded and error state
+  const [state, setState] = useState({
+    loaded: false,
+    error: false
+  });
+
+  useEffect(
+    () => {
+      // If cachedStyleSheets array already includes href that means another instance ...
+      // ... of this hook already loaded this script, so no need to load again.
+      if (cachedStyleSheets.includes(href)) {
+        setState({
+          loaded: true,
+          error: false
+        });
+      } else {
+        cachedStyleSheets.push(href);
+
+        // Create stylesheet link
+        let linkElement = document.createElement('link');
+        linkElement.rel = 'stylesheet';
+        linkElement.href = href;
+        linkElement.media = media;
+
+        // Stylesheet event listener callbacks for load and error
+        const onStyleSheetLoad = () => {
+          setState({
+            loaded: true,
+            error: false
+          });
+        };
+
+        const onStyleSheetError = () => {
+          // Remove from cachedStyleSheets we can try loading again
```

Comment:
Hmmm not sure I follow, what if it was a temporary network error?

Reply:
It can be a temporary error. But it can also happen that it's not temporary (or temporary but takes like 1 hour). Then it would keep sending requests on every render.

Reply:
Which is probably not a problem for the timeline block, but could be for blocks that render more frequently.